geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/filters/CqlTransformFilter.scala

```diff
@@ -85,30 +94,39 @@ object CqlTransformFilter extends StrictLogging {
     * @param sft simple feature type
     * @param filter cql filter, if any
     * @param transform transform, if any
+    * @param hints is used to get sampling options
     * @return
     */
   def apply(
       sft: SimpleFeatureType,
       index: GeoMesaFeatureIndex[_, _],
       filter: Option[Filter],
-      transform: Option[(String, SimpleFeatureType)]): CqlTransformFilter = {
-    if (filter.isEmpty && transform.isEmpty) {
+      transform: Option[(String, SimpleFeatureType)],
+      hints:Hints): CqlTransformFilter = {
+    if (filter.isEmpty && transform.isEmpty && hints.getSampling.isEmpty) {
       throw new IllegalArgumentException("The filter must have a predicate and/or transform")
     }
 
     val feature = KryoFeatureSerializer(sft, SerializationOptions.withoutId).getReusableFeature
     feature.setIdParser(index.getIdFromRow(_, _, _, null))
     transform.foreach { case (tdefs, tsft) => feature.setTransforms(tdefs, tsft) }
 
+    val samplingOptions: Option[(Float, Option[String])] = hints.getSampling
+
     val delegate = filter match {
-      case None => new TransformDelegate(sft, index, feature)
-      case Some(f) if transform.isEmpty => new FilterDelegate(sft, index, feature, f)
-      case Some(f) => new FilterTransformDelegate(sft, index, feature, f)
+      case None if samplingOptions.isDefined && transform.isEmpty => new FilterDelegate(sft, index, feature, Filter.INCLUDE,samplingOptions)
+      case None if samplingOptions.isDefined => new FilterTransformDelegate(sft, index, feature, Filter.INCLUDE,samplingOptions)
+      case None => new TransformDelegate(sft, index, feature,samplingOptions)
+      case Some(f) if transform.isEmpty => new FilterDelegate(sft, index, feature, f,samplingOptions)
+      case Some(f) => new FilterTransformDelegate(sft, index, feature, f,samplingOptions)
```

Comment:
this logic statement is getting a little hairy, it might be nice to clean it up a little, maybe make it `(filter, transform, samplingOptions) match {`